File: Jack_Charris/Programas/Dart/entero.dart
import 'dart:io';

void main(){

    int? n;
    do {
        print('Ingresa un numero entero: '); 

        n = int.tryParse(stdin.readLineSync()!);

       if ( n == null ){
         print(" Ingrese un numero entero valido "); 
       }
           
    } while ( n == null );

    if ( n > 0 ){
        print("El numero es positivo");
    } else if ( n < 0 ){
        print("El numero es negativo");
    } else {
        print("El numero ingresado es cero");
    }
     
}

File: Jack_Charris/Programas/Ruby/entero.rb
def main
    n = nil
    loop do
      print 'Ingresa un numero entero: '
      input = gets.chomp
  
      # Intentar convertir la entrada a un entero
      n = Integer(input) rescue nil
  
      if n.nil?

            puts 'Ingrese un numero entero valido.'
        
      else
         break
      end

    end
  
    # Verificar si el número es cero, positivo o negativo
    if n > 0

      puts 'El numero es positivo.'

    elsif n < 0

      puts 'El numero es negativo.'

    else

      puts 'El numero ingresado es cero.'

    end

  end
  
  main